metaflow/mflog/__init__.py

```diff
@@ -41,21 +41,23 @@
     " }" % TASK_LOG_SOURCE
 )
 
-BASH_SAVE_LOGS_ARGS = ["python", "-m", "metaflow.mflog.save_logs"]
-BASH_SAVE_LOGS = " ".join(BASH_SAVE_LOGS_ARGS)
+
+def _get_bash_capture_log(default_python_executable="python"):
+    bash_save_log_args = [default_python_executable, "-m", "metaflow.mflog.save_logs"]
+    return bash_save_log_args, " ".join(bash_save_log_args)
 
 
 # this function returns a bash expression that redirects stdout
 # and stderr of the given bash expression to mflog.tee
-def bash_capture_logs(bash_expr, var_transform=None):
+def bash_capture_logs(bash_expr, var_transform=None, default_python_executable_path="python"):
```

Comment:
nit: consistency: for bootstrap, it defaults to none but here to python. We can make it python for both or none for both.